torch/overrides.py

```diff
@@ -1905,8 +1905,11 @@ def get_buffer(tensor_subclass, data, prefix):
     import ctypes
     assert prefix in {"stride", "size", "sym_size"}
     buffer_name = f"_{prefix}_buffer"
-    if not hasattr(tensor_subclass, buffer_name):
-        SizeType = ctypes.c_longlong * len(data)
-        setattr(tensor_subclass, buffer_name, SizeType(*data))
+    # We explicitly do *not* cache.
+    # (1) caching sizes/strides here means that the cache is stale if we update metadata
+    # (2) caching doesn't save us much anyway - we have to go into python every time
+    #     the user calls .shape in any case.
+    SizeType = ctypes.c_longlong * len(data)
+    setattr(tensor_subclass, buffer_name, SizeType(*data))
```

Comment:
@bdhirsh  I ended up needing to fix this as well to get NestedTensor to properly return a singleton symint through the slow path. 
So in case you haven't gotten the chance to look at this further, I have a PR https://github.com/pytorch/pytorch/pull/107839.

The PR removes `values_from_buffer` as you suggested, but also account for Ed's comment https://github.com/pytorch/pytorch/pull/106405/files#r1282302973